### DISCUSSION TITLE
Option to get started without Flux CLI

### DISCUSSION THREAD

**Discussion by Thaval** [fluxcd/flux2#2741] [Q&A] [answered]
Hey, 

is there an option to get my manifests deployed without running "flux create...."? 
The problem is that the cli is not as powerful as the manifest yaml and so comes that I need to adjust the yaml a lot and I want to avoid deploying the resources twice as it can be error prone.

**Comment by somtochiama** [comment]
You can run `flux create --export`, edit the yaml and then apply it using `kubectl` or push to git?
Or are you talking about adjusted the manifested generated by `flux bootstrap`?

**Comment by Thaval** [comment, marked as answer]
That's what I'm currently doing. Editing the yaml and then push them back to git. kubectl apply -f does not work as it cannot find some schemas. However, I'd find it better if I could just apply my manifest files, this would ease my logic.

**Reply by souleb** [reply]
`flux install` install the crds for you. Then it's up to you if you want to write your own yamls instead of relying on `flux create`. 

You can find the installed crds here: https://github.com/fluxcd/flux2/blob/main/manifests/crds/kustomization.yaml, if you wish to install them yourself

**Reply by Thaval** [reply]
Thanks for the response! 
These components are already installed. 
What I want to do is creating the HelmRelease, Kustomization and GitRepository by running "kubectl apply -f ...". 
Currently I'm running "flux create..", export these resources, edit the yaml and push them. I'd like to do yaml templating, so copy the templates, exchange the values I want and then simply run "kubectl apply -f ...". This is a much more cleaner approach imho.

Will this even work? Dunno what "flux create..." is doing under the hood.

Edit: Tested it out running `kubectl apply -k .`. The resources (HelmRelease, GitRepository, HelmChart) is created, but the GitRepository is throwing an error: 
`failed to get secret: Secret not found`

**Reply by souleb** [reply]
`flux create` is a helper, with it you do not have to write your own yaml files. With `--export` flag, it outputs what it would have applied on your cluster using your `kubeconfig`. Now the error here is most likely due to the fact that your `gitrepository` declaration references a secret. So you should have an existing secret in the cluster.

**Reply by Thaval** [reply]
Yea, that's true. 
I got it. I created a wrong Base64 string for the username and password in the secret. So I just copy pasted whatever flux did to encode these values and now it seems to work. 
Thanks for your help.

PS: Testing if it is updated as expected.

**Reply by Thaval** [reply]
Well it work partly. The GitRepository, HelmRelease and Kustomization do not reconcile when I apply the resources with "kubectl apply -k ."
